### DISCUSSION TITLE
How to use Nested Rest API Request with dynamic variables ?

### DISCUSSION THREAD

**Discussion by gdespres** [Budibase/budibase#8441] [Help] [answered]
Hi everybody, I don't understand when the dynamic variable content is updated.

I need to use 2 nested API Rest requests: 
- The first one search a user by {{ Current User.email }}
<img width="1673" alt="Capture d’écran 2022-10-29 à 18 23 55" src="https://user-images.githubusercontent.com/2405993/198842624-30a25004-bc77-4204-85be-a101f054a027.png">
<img width="1676" alt="Capture d’écran 2022-10-29 à 18 24 05" src="https://user-images.githubusercontent.com/2405993/198842707-2a618ac0-2a1c-43c0-bbdc-a80b24cf7f1b.png">

- The Second one get the current User (logged as the previous user.id via Dynamic Variable)
<img width="1668" alt="Capture d’écran 2022-10-29 à 18 35 09" src="https://user-images.githubusercontent.com/2405993/198842885-a645b092-b020-4188-adc7-ccc1290ceec3.png">

When I use these 2 providers In design mode, the dynamic variable is not updated with the result of the first request.
<img width="1678" alt="Capture d’écran 2022-10-29 à 18 28 39" src="https://user-images.githubusercontent.com/2405993/198842782-a6216f4e-c9e9-4d8b-b472-84a10110fc25.png">

What i'm doing wrong ?
Thanks

**Comment by deanhannigan** [comment, marked as answer]
Hi @gdespres, 

Dynamic variables in REST requests will currently only update in 2 possible scenarios: 

1. The dynamic variables current value expires. This expiry is currently a period of `1 hour` from the point the variable was last stored.
2. The query making use of the variable fails with a response of 400 or greater. In this case, if `Get Current User` fails when it is executed, the query will attempt to update the dynamic variables in `Search by User Email` in order to possibly allow the request to succeed.

Also, in this instance it isn't strictly necessary to nest your datasources in the builder. Your dynamic variable is stored against the `GitLab` datasource and will be available to any sibling queries.

If you have any further questions, let us know!
